### PR TITLE
Update secrets fetching / var names. Closes #57.

### LINKED ISSUE #57: Update secrets fetching / var names.
To work with the new format in `/var/mnt/secrets/rds-bocoup-db`.

### PR DESCRIPTION
**Pull request by cowboy**
Bonus: add `db:create` and `db:drop` npm scripts.

TODO:

Update [Development wiki page](https://github.com/bocoup/skillsbot/wiki/Development) to mention:
- Creating a db with `npm run db:create`
- Dropping a db with `npm run db:drop`
- The updated "Add Bocoup db secrets" command, which is now:
  - `ssh nest.bocoup.com cat /mnt/secrets/rds-bocoup-db/db/bocoup | sed 's/^PG/BOCOUP_DB_/' >> .env`

### CODE REVIEW

**Comment by cowboy** [comment]
I have a few concerns about this, I'll post them separately.

### Concern 1

I don't like that this is creating an `.env` file that looks like:

```
SLACK_CLIENT_ID=...
SLACK_CLIENT_SECRET=...
PGHOST=...
PGPORT=...
PGUSER=skillsbot
PGNAME=skillsbot
PGPASSWORD=...
```

I'd like the `PGNAME` to be `skillsbot-dev` by default, to possibly prevent people from accidentally running the dev bot against the production database.

I could do this in the npm script, to add a `-dev` to the end of the database user, but I don't know if it's enough.

```
ssh nest.bocoup.com cat /mnt/secrets/{pombot-slack,rds-bocoup-db/{connection,db/pombot}} | sed -E 's/^(PGNAME=.*)/\1-dev/'
```

What do you think about creating a separate user for production `skillsbot` and development `skillsbot-dev`, each with their own password and corresponding secrets file in `/mnt/secrets/rds-bocoup-db/db/`, and using the same pattern for all bot-related databases?

To recap, what I'm proposing is having secrets like the following:

**`/mnt/secrets/rds-bocoup-db/db/skillsbot`**

This file would be referenced by the ansible config to get secrets for production. The `skillsbot` user wouldn't need to be able to create or drop databases. There would just be the one production database.

```
PGUSER=skillsbot
PGNAME=skillsbot
PGPASSWORD=...
```

**`/mnt/secrets/rds-bocoup-db/db/skillsbot-dev`**

This file would be referenced in the npm script to get secrets for development. Per-user dev databases would be created and dropped via the `skillsbot-dev` user.

```
PGUSER=skillsbot-dev
PGNAME=skillsbot-dev
PGPASSWORD=...
```

**Comment by cowboy** [comment]
### Concern 2

I think the npm `get-secrets` script should pull down a different set of app slack secrets for development than the ansible config does for deploying to production.

To that end, I think we should have secrets like the following:

**`/mnt/secrets/skillsbot-production`**

This file would be referenced by the ansible config to get secrets for production. It would contain the actual production slack app secrets.

```
SLACK_CLIENT_ID=...
SLACK_CLIENT_SECRET=...
```

**`/mnt/secrets/skillsbot-dev`**

This file would be referenced in the npm script to get secrets for development, and would have no values for the secrets. This allows (forces) the user to fill in the secrets from their own app, since everyone will need their own app, with a different bot name, different callback urls, etc.

```
SLACK_CLIENT_ID=
SLACK_CLIENT_SECRET=
```

**Comment by cowboy** [comment]
@tkellen: thoughts?

**Comment by tkellen** [comment]
Looking now.

**Comment by tkellen** [comment]
I'm :+1: to the idea generally. I have a few questions that would be better served by answering in Slack.

File: package.json [hunk 1]
```diff
@@ -3,8 +3,10 @@
   "private": true,
   "version": "0.0.1",
   "scripts": {
-    "get-secrets": "ssh nest.bocoup.com \"cat /mnt/secrets/{rds,skillsbot-slack,skillsbot-db-dev}\" > ./.env",
-    "db": "source .env && PGPASSWORD=$DB_PASSWORD psql -h $DB_HOST -p $DB_PORT -U $DB_USER $DB_NAME",
+    "get-secrets": "ssh nest.bocoup.com cat /mnt/secrets/{skillsbot/slack-dev,rds-bocoup-db/{connection,db/skillsbot-dev}} > .env",
+    "db": "export $(<.env) && psql $PGNAME",
+    "db:create": "export $(<.env) && createdb $PGNAME",
+    "db:drop": "export $(<.env) && dropdb -i $PGNAME",
```
Comment (tkellen) [inline package.json:9]: this scares me a bit but as long as we can't accidentally do this to the production database, awesome

**Comment by tkellen** [comment]
:+1:

File: package.json (hunk 1 above)
Comment (cowboy) [inline package.json:9]: FWIW, the `-i` flag to `dropdb` creates a confirmation prompt.

Comment (tkellen) [inline package.json]: /me nods.
